### ISSUE TITLE
Fix ResourceWarnings from unclosed files

### ISSUE THREAD

**Issue by jacobtylerwalls** [archesproject/arches#11091]
Running the tests with `ResourceWarning` enabled flags many instances of unclosed file handles, which is a performance consideration.

We should rewite these `open` calls as `with open(...`:

### Reproducer
```sh
python3 -Wall::ResourceWarning manage.py test tests --settings=tests.test_settings --verbosity=2
```

### Results
Items like:
```py
test_csv_export (tests.exporter.resource_export_tests.BusinessDataExportTests.test_csv_export) ... /Users/jwalls/prj/arches/arches/app/utils/data_management/resources/importer.py:111: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.mapping' mode='r' encoding='UTF-8'>
  self.mapping = json.load(open(path, "r"))
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/Users/jwalls/prj/arches/tests/exporter/resource_export_tests.py:73: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.csv' mode='r' encoding='utf-8'>
  BusinessDataImporter(
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/Users/jwalls/prj/arches/arches/app/utils/data_management/resources/formats/csvfile.py:73: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.mapping' mode='r' encoding='UTF-8'>
  resource_export_configs = json.load(open(configs, "r"))
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/Users/jwalls/prj/arches/tests/exporter/resource_export_tests.py:87: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.csv' mode='r' encoding='utf-8'>
  csv_input = list(
ResourceWarning: Enable tracemalloc to get the object allocation traceback
...
```

**Comment by HarisMughal** [comment]
Hi, can I work on this?

**Comment by chiatt** [comment]
Hi @HarisMughal - you're definitely welcome to work on this.

**Comment by jacobtylerwalls** [comment]
Hey @HarisMughal :wave: thanks for the message. When getting started, I'd suggest branching off of dev/7.6.x instead of master.

**Comment by jacobtylerwalls** [comment]
Fixed in #11108
